modules/kubernetes-addons/cert-manager/locals.tf

```diff
@@ -21,22 +21,13 @@ locals {
     var.helm_config
   )
 
-  set_values = [
```

Comment:
@askulkarni2 / @vara-bonthu - As aws-privateca-issuer depends on cert manager, i received the following error in cert-manager pod, which was resulting in certificate requests getting failed to be mounted as secrets.

```
E0322 01:59:36.653678       1 controller.go:163] cert-manager/certificaterequests-approver "msg"="re-queuing item due to error processing" "error"="admission webhook \"webhook.cert-manager.io\" denied the request: status.conditions: Forbidden: user \"system:serviceaccount:cert-manager:cert-manager-sa\" does not have permissions to set approved/denied conditions for issuer {aws001-preprod-dev-eks AWSPCAClusterIssuer awspca.cert-manager.io}" "key"="default/example-nftjp" 
```

Helm chart installation of cert-manager would pass and pods kept running, but when used by aws-privateca-issuer or any other Issuer, it would log errors.

On digging further, i realized that these two parameters were culprit and was incorrectly set, resulting in cert-manager RBAC to not work as it just created a SA without any permissions as what the helm chat typically would do. IRSA is not required here as its the PCA pod , which uses IRSA. [cert-manager](https://cert-manager.io/docs/installation/helm/) helm chart guide also has no such requirements.

Reply:
Can we confirm that we do not need IRSA for any other use cases? If not then `irsa_config` can be removed altogether and just create the namespace instead such as [here](https://github.com/aws-samples/aws-eks-accelerator-for-terraform/blob/dbcb4fe9e5da10032f30dea2ed3f3946808d7c0e/modules/kubernetes-addons/ingress-nginx/main.tf#L11).

Reply:
@anshrma Its because Private CA add-on is expecting the  cert-manager **service account name** as `cert-manager`

Reply:
@askulkarni2 Yes, we need IRSA for cert-manager

Reply:
with a policy like
```json
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Sid": "AllowRoute53",
            "Effect": "Allow",
            "Action": [
                "route53:ChangeResourceRecordSets",
                "route53:ListResourceRecordSets"
            ],
            "Resource": [
                "arn:aws:route53:::hostedzone/${hosted_zone_id}"
            ]
        },
        {
            "Sid": "AllowRoute53-GetChange",
            "Effect": "Allow",
            "Action": [
                "route53:GetChange"
            ],
            "Resource": [
                "arn:aws:route53:::change/*"
            ]
        },
        {
            "Sid": "AllowRoute53-ListHostedZonesByName",
            "Effect": "Allow",
            "Action": [
                "route53:ListHostedZonesByName"
            ],
            "Resource": [
                "*"
            ]
        }
    ]
}
```